tests/integration/test_charm.py

```diff
@@ -17,22 +17,25 @@ async def test_build_and_deploy(ops_test: OpsTest):
     openldap_charm = await ops_test.build_charm(".")
     await asyncio.gather(
         ops_test.model.deploy("postgresql-k8s", application_name=PSQL, num_units=1),
-        ops_test.model.deploy(openldap_charm,
-                              resources={'openldap-image': 'openldapcharmers/openldap:2.4.50'},
-                              application_name=APP_NAME, num_units=1),
+        ops_test.model.deploy(
+            openldap_charm,
+            resources={'openldap-image': 'openldapcharmers/openldap:2.4.50'},
```

Comment:
I'm a little concerned about the fact that the image and the charm do not share the lifecycle. It think it is error-prone in this case.
If the images are also tagged as latest maybe it'd be worth to use that latest tag here (or latest version 2 tag if that exists).

Reply:
Yes, that is a good point for discussion. There are things I'd like to get opinions on:
* should we share the lifecycle of charms and images when possible - I mean if an app is stable and doesn't change often, but we add features to the charm, won't that naturally drift?
* should we try to own the images of the charm - if there is an openldap image that is available (and official/acclaimed/has X amount of stars etc.), should we use it or should we build our own? Building our own means more control and more responsibility (security fixes, maintenance)
* should we test against stable and latest - If I get your comment right, we can build the image, push it locally to microk8s registry and test against latest. That is good and I like it if we control the image. Even if we don't, we might want to containerize the app as :latest and run a test against it. Specifically on the point of using latest, I'd say we don't do it - a charm needs to be versioned and all the dependencies of that charm need to be versioned as well imo. So an openldap-k8s charm version v0.23.2 should work and be tested against openldap:2.54.1 for instance.

What do you think?

Reply:
Well, in my opinion:
- We should share the lifecycle of charms and images whenever possible, as it simplifies version control and delivery. I don't see any reason to increase complexity by splitting them.
- I don't think it makes sense to reinvent the wheel, but maybe there's some reason I'm missing
- I agree with with your point of having repeatable builds, however, the charm teams are currently expecting this not to be the case. Until we reach a compromise, what do you think about reading the image tag from the metadata to have it defined in a single point?

Reply:
Good points, thanks for the detailed feedback! It makes sense to own the image, as we can develop it hand in hand with the charm and share the lifecycle. 
Specifically for this charm, do you have a suggestion - it can be built and tested here (so docker build, push to microk8s registry, test it the same way we do with Indico charm), question is where to push it?
I think you are right with reading the image tag and to have it defined in a single point, that is definetely the way to go and I'll change that. 
Thanks again for the comments!

Reply:
For Indico, we're pushing it directly into charmhub as a resource. Unfortunately, it is not very straightforward for now and certainly not very fast as you can see in the [this run](https://github.com/canonical/indico-operator/actions/runs/2873522585)
Hopefully the existing tooling will evolve to better support this scenario.

Reply:
Good, this can be the right approach. I think it's better to formulate this as an issue, as the scope is getting broader and I'm working on this as part of an onboarding story. Are you fine with me creating an issue about the image and resolve this there? Thanks for the pointers btw!

Reply:
Yes, sure!

Reply:
https://github.com/canonical/openldap-k8s-controller/issues/2 created

Reply:
Note that I've adjusted the integration test and now the resource is inferred from the `metadata.yaml` in a fashion similar to how the file is parsed from the `pytest operator`. So a change in the metadata.yaml will be reflected here. With solving #2 we can iterate over the approach.